packages/rocketchat-lib/server/methods/setName.coffee

```diff
@@ -0,0 +1,19 @@
+Meteor.methods
+	setName: (name) ->
+		if not Meteor.userId()
+			throw new Meteor.Error('invalid-user', "[methods] setName -> Invalid user")
+
+		console.log '[methods] setName -> '.green, 'userId:', Meteor.userId(), 'arguments:', arguments
+
+		user = Meteor.user()
+
+		if user.name is name
+			return name
+
+		if not /^[0-9a-zA-Z-_. ]+$/.test name
```

Comment:
Name should allow the real user name, so I think only trim the name and verify is empty is the correct way